api/websocket/server/server.go

```diff
@@ -109,8 +104,23 @@ func (ws *WsServer) Start() error {
 	ws.server = &http.Server{Handler: http.HandlerFunc(ws.websocketHandler)}
 	go ws.server.Serve(ws.listener)
 
-	ws.tlsServer = &http.Server{Handler: http.HandlerFunc(ws.websocketHandler)}
-	go ws.tlsServer.Serve(ws.tlsListener)
+	go func(wssCertReady chan struct{}) {
+		for {
+			timer := time.NewTimer(time.Second * 5)
+			select {
+			case <-wssCertReady:
+				log.Info("wss cert ready")
+				ws.tlsListener, err = ws.initTlsListen()
+				if err != nil {
+					log.Error("Https Cert: ", err.Error())
+				}
+				ws.tlsServer = &http.Server{Handler: http.HandlerFunc(ws.websocketHandler)}
```

Comment:
Why websocket separate `ws.server` and `ws.tlsServer` while http only has a single `s.httpServer`?